app/main.py

```diff
@@ -1,2 +1,53 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.table_size = 8
+        self.hash_table: list = [None] * self.table_size
+
+    def __setitem__(
+            self,
+            key: (int, float, bool, tuple, str),
+            value: Any
+    ) -> None:
+        hashed_value = hash(key)
+        index = hashed_value % self.table_size
+        while self.hash_table[index] is not None:
+            saved_key, saved_hashed_value, saved_value = self.hash_table[index]
+            if saved_key == key and hashed_value == saved_hashed_value:
+                self.hash_table[index] = (key, hashed_value, value)
+                break
+            print(f"The key {key} collided with {self.hash_table[index]}")
+            index = (index + 1) % len(self.hash_table)
+        if self.hash_table[index] is None:
+            self.hash_table[index] = (key, hashed_value, value)
+            self.length += 1
+            self.resize()
+
+    def __getitem__(
+            self,
+            key: (int, float, bool, tuple, str)
+    ) -> Any:
+        hashed_value = hash(key)
+        index = hashed_value % self.table_size
+        while True:
+            try:
```

Comment:
do not use `try` `except` where you can just check if the cell is empty.
it makes the code less readable